src/mlpack/tests/load_save_test.cpp

```diff
@@ -768,6 +769,33 @@ BOOST_AUTO_TEST_CASE(SaveHDF5Test)
 
 #endif
 
+
+/**
+ * Test dictionary encoding.
+ */
+BOOST_AUTO_TEST_CASE(DictionaryEncodingTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "hello how are you";
+  arr[1] = "i am good";
+  arr[2] = "Good how are you";
+  std::unordered_map<string, size_t>mappings;
+  arma::mat output;
+  data::Encode(arr, mappings, output);
+  BOOST_REQUIRE_EQUAL(output(0, 0), 1);
+  BOOST_REQUIRE_EQUAL(output(0, 1), 2);
+  BOOST_REQUIRE_EQUAL(output(0, 2), 3);
+  BOOST_REQUIRE_EQUAL(output(0, 3), 4);
+  BOOST_REQUIRE_EQUAL(output(1, 0), 5);
+  BOOST_REQUIRE_EQUAL(output(1, 1), 6);
+  BOOST_REQUIRE_EQUAL(output(1, 2), 7);
+  BOOST_REQUIRE_EQUAL(output(1, 3), 0);
+  BOOST_REQUIRE_EQUAL(output(2, 0), 8);
+  BOOST_REQUIRE_EQUAL(output(2, 1), 2);
+  BOOST_REQUIRE_EQUAL(output(2, 2), 3);
+  BOOST_REQUIRE_EQUAL(output(2, 3), 4);
```

Comment:
Technically when testing like this, we should simply be ensuring that each string maps to *something*.  We don't actually care that `hello` maps to 1, we care that `hello` maps to something different than `how`.